### PR TITLE
TrekkieRun Delete Endpoint

### PR DESCRIPTION
**Pull request by revol-xut**
# Things Done

Adding endpoint for deleting trekkie runs.

### CODE REVIEW

File: src/routes/trekkie_runs.rs [hunk 1]
```diff
@@ -133,3 +141,49 @@ pub async fn trekkie_run_update(
...
+        Err(e) => {
+            error!("cannot get connection from connection pool {:?}", e);
+            return Err(ServerError::InternalError);
+        }
+    };
+
+    let user_session = fetch_user(identity, &mut database_connection)?;
+
+    if !user_session.is_admin() {
+        return Err(ServerError::Unauthorized);
```
Comment (oxapentane) [inline src/routes/trekkie_runs.rs:173]: If user is authed and does not have enough privs should be 403 Forbidden

**Review by oxapentane — CHANGES_REQUESTED** [review]
One small nitpick regarding response code, but in general can be merged

File: src/routes/trekkie_runs.rs (hunk 1 above)
Comment (revol-xut) [reply src/routes/trekkie_runs.rs]: done

File: src/routes/trekkie_runs.rs [hunk 2]
```diff
@@ -50,6 +51,13 @@ pub async fn trekkie_run_list(
         }
     };
 
+    // fetch user session
+    let session_user = fetch_user(identity, &mut database_connection)?;
+
+    if !session_user.is_admin() {
+        return Err(ServerError::Unauthorized);
```
Comment (oxapentane) [inline src/routes/trekkie_runs.rs:58]: Slipped through the cracks :)

File: src/routes/auth.rs [hunk 3]
```diff
@@ -120,13 +120,13 @@ pub async fn user_login(
...
-                debug!("Password does not match");
+                debug!("passwords dont match");
                 Err(ServerError::BadClientData)
             }
         }
         Err(e) => {
-            error!("Err: {:?}", e);
-            Err(ServerError::InternalError)
+            error!("user with this email not found or postgres error: {:?}", e);
+            Err(ServerError::Unauthorized)
```
Comment (oxapentane) [inline src/routes/auth.rs:129]: Couple of possible problems with this: First, it leaking user emails by giving different response on invalid email or password.

Second (more of a nitpick): If we have a genuine Postgres fucky, this is Internal Server Error.

**Review by oxapentane — CHANGES_REQUESTED** [review]

**Pull request merged**